docs/edgeql/expressions/parameters.rst

```diff
@@ -0,0 +1,173 @@
+.. _ref_eql_expr_params:
+
+Parameters
+==========
+
+:edb-alt-title: Query Parameters
+
+A parameter reference is used to indicate a value that is supplied externally
+to an EdgeQL expression.
+
+An example of query, that can be parametrized:
+
+.. code-block:: edgeql-repl
+
+    db> SELECT Person { birth_date } FILTER .first_name = 'John';
+    {
+        Object {birth_date: <cal::local_date>'1971-01-01'},
+    }
+
+The equivalent parametrized query:
+
+.. code-block:: edgeql-repl
+
+    db> SELECT Person { birth_date } FILTER .first_name = <str>$name;
+    Parameter <str>$name: John
+    {
+        Object {birth_date: <cal::local_date>'1971-01-01'},
+    }
+
+Note how the actual parameter value is asked by REPL in a separate prompt,
+after the whole query is accepted.
+
+While parameters are useful in REPL, their power is mostly when using language
+bindings. Example of using parameters in Python:
+
+.. code-block:: python
+
+    await connection.fetchone(
+        "SELECT Person { birth_date } FILTER .first_name = <str>$name",
+        name="John")
+
+Example of using parameters in JavaScript:
+
+.. code-block:: javascript
+
+    await connection.fetchOne(
+        "SELECT Person { birth_date } FILTER .first_name = <str>$name",
+        {name: "John"})
+
+
+Parameter Types
+---------------
+
+The following values can be turned into parameters:
+
+* Any :ref:`scalar type <ref_datamodel_scalar_types>`
+* :eql:type:`Array <array>` of any scalar type
+
+Parameter type is specified in the angular brackets in front of variable
+name, similarly to normal type cast:
+
+.. code-block:: edgeql
+
+    INSERT Person {
+        first_name := <str>$first_name,
+        age := <int32>$age,
+        interests := <array<str>>$interests,
+    }
+
+See :ref:`ref_eql_types` for more info on specifying types.
+
+
+Optional Parameters
+-------------------
+
+By default parameter types are ``REQUIRED`` which means that query will fail
+if parameter value is nil. You can use ``OPTIONAL`` keyword for optional
+arguments.
+
+Example in Python:
+
+.. code-block:: python
+
+    for name, age in [("John", 33), ("Jack", None)]:
+        await conn.fetchone("""
+            INSERT Person {
+                first_name := <str>$name,
+                age := <OPTIONAL str>$age,
+            }
+        """, name=name, age=age)
+
+Example in JavaScript:
+
+.. code-block:: javascript
+
+    for(let [name, age] of [("John", 33), ("Jack", null)]) {
+        await conn.fetchOne("""
+            INSERT Person {
+                first_name := <str>$name,
+                age := <OPTIONAL str>$age,
+            }
+        """, {name, age})
+    }
+
+Note: there is a language-specific nil value, ``None`` in Python example and
```

Comment:
We don't use the "nil" term anywhere in the docs and IMO we shouldn't introduce it here.

```suggestion
In most languages there is a way to indicate "absence of value", e.g. in Python we use ``None``, and we use
```

Reply:
Yes. We don't have a term for that. And this is a problem. This one waits for fixing https://github.com/edgedb/edgedb/pull/1373#pullrequestreview-403197099

Reply:
Alright, see the below suggestion.